### PR TITLE
Don't assume $0 is the script source

### PR DESCRIPTION
**Pull request by chmouel**
# Changes

Don't assume $0 is the script source

Part of our 'downstream' CI we reuse the `test/e2e-common.sh` functions.

We have a script that source the test/e2e-common.sh and run with the downstream
specifics functions.

$0 in the e2e-common assume the script is run from `test/e2e-tests.sh` when our
script which source test/e2e-tests.sh is something else somewhere else.

Let use git to safely detect the toplevel of the directory from where we can
source the vendor directory.

Remove latest go get from there too

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [🤷🏼‍♂️] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [🤷🏼‍♂️] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [👍] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
## Reviewer Notes

# Release Notes

### CODE REVIEW

File: test/e2e-tests-yaml.sh [hunk 1]
```diff
@@ -17,8 +17,7 @@
 # This script calls out to scripts in tektoncd/plumbing to setup a cluster
 # and deploy Tekton Pipelines to it for running integration tests.
 
-go get -d github.com/GoogleCloudPlatform/cloud-builders/gcs-fetcher/cmd/gcs-fetcher
```
Comment (ImJasonH) [inline test/e2e-tests-yaml.sh]: Is this line no longer necessary? Was it ever?

I think I'd prefer to keep it unless we fully understand why it's here, and remove it in a separate PR.

cc @wlynch

Comment (vdemeester) [reply test/e2e-tests-yaml.sh]: @ImJasonH I also have no idea why it was there. Also, as we are still using the `vendor` folder, the code is already there in the `vendor` folder :angel:

Comment (chmouel) [reply test/e2e-tests-yaml.sh]: and having this will re-download the whole `vendor/` directory.....

Comment (vdemeester) [reply test/e2e-tests-yaml.sh]: well it depends on the cache state, but it will make it download stuff that are already there

Comment (ImJasonH) [reply test/e2e-tests-yaml.sh]: Yeah I’m not convinced we need it but I’d just rather remove it in a deliberate separate change.

Comment (chmouel) [reply test/e2e-tests-yaml.sh]: I have removed that change and separated to two commits..

**Pull request merged**